java/src/jmri/ThrottleListener.java

```diff
@@ -38,67 +63,113 @@
      * @param reason  The reason why the throttle request failed.
      */
     public void notifyFailedThrottleRequest(LocoAddress address, String reason);
-
+    
     /**
-     * Get notification that a throttle request requires is in use by another
-     * device, and a "steal" may be required.
+     * Get notification that a throttle request is in use by another
+     * device, and a "steal", "share", or "steal/share" decision may be required.
      *
-     * @param address LocoAddress of the throttle that needs to be stolen.
+     * @param address The LocoAddress that needs the decision.
+     * @param question The question being asked, steal / cancel, share / cancel, steal / share / cancel
      */
-    public void notifyStealThrottleRequired(LocoAddress address);
+    public void notifyDecisionRequired(LocoAddress address, DecisionType question);
 
 }
 
 /*
  * @startuml jmri/doc-files/ThrottleListener-Sequence.png
- * participant ThrottleListener
- * participant ThrottleManager
- * participant Throttle
```

Comment:
I may have told you to remove this at some point, but we probably should put the throttle back in the diagram, at least for the new throttle request and return.

Reply:
Resolved thanks to reworking of chart